src/services/codefixes/inferFromUsage.ts

```diff
@@ -674,17 +671,49 @@ namespace ts.codefix {
             }
         }
 
+        interface Priority {
+            high: (t: Type) => boolean;

```

Comment:
So, something can be high and low, or neither high nor low?

Reply:
Yes. I'm not sure priority is the right name, and not sure whether high/low are the right names. Ideas?

Reply:
To answer your question, an inference can be high, low or neither. If there is a mistake in the specification, it can be low and high.  I'll add an assert.

Reply:
Maybe this could be one function returning an enum.

Reply:
Hm. I tried it and it simplified removeLowPriorityInferences a bit, made the vacuous/non-vacuous check simpler, and made the other two checks harder to read. I think it's a wash and I prefer the interface structure.